airbyte-metrics/src/main/java/io/airbyte/metrics/DogstatsdMetricSingleton.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics;
+
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
+import com.timgroup.statsd.StatsDClient;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+
+/**
+ * Light wrapper around the DogsStatsD client to make using the client slightly more ergonomic.
+ * <p>
+ * This class mainly exists to help Airbyte instrument/debug application on Airbyte Cloud.
+ * <p>
+ * Open source users are free to turn this on and consume the same metrics.
+ */
+@Slf4j
+public class DogstatsdMetricSingleton {
+
+  private static DogstatsdMetricSingleton instance;
+  private final StatsDClient statsDClient;
+  private final boolean instancePublish;
+
+  public DogstatsdMetricSingleton(final String appName, final boolean publish) {
+    instancePublish = publish;
+    statsDClient = new NonBlockingStatsDClientBuilder()
+        .prefix(appName)
+        .hostname(System.getenv("DD_AGENT_HOST"))
+        .port(Integer.parseInt(System.getenv("DD_DOGSTATSD_PORT")))
+        .build();
+  }
+
+  public static synchronized DogstatsdMetricSingleton getInstance() {
+    if (instance == null) {
+      throw new RuntimeException("You must initialize configuration with the initializeMonitoringServiceDaemon() method before getting an instance.");
+    }
+    return instance;
+  }
+
+  public synchronized static void initialize(final String appName, final Map<String, String> mdc, final boolean publish) {
```

Comment:
Why is the MDC important here? Shouldn't the initialization be run from some thread that has the MDC configured?

Reply:
good point - I think it's left over from when I was testing. You should be right that we don't need this anymore. Will remove.